kahuna/public/js/services/image/usages.js

```diff
@@ -0,0 +1,60 @@
+import angular from 'angular';
+import Immutable from 'immutable';
+
+import Rx from 'rx';
+
+export const imageUsagesService = angular.module('gr.image-usages.service', ['kahuna.edits.service']);
+
+
+imageUsagesService.factory('imageUsagesService', [function() {
+
+    function forImage(imageResource) {
+        function usageTitle(usage) {
+            const referenceType =
+                usage.get('platform') == 'print' ? 'indesign' : 'frontend';
+
+            const build = (usage, referenceType) => {
+                const reference = usage.get('references').find(u =>
+                    u.get('type') == referenceType);
+
+                return (reference && reference.get('name')) ? reference.get('name') : 'No title found.';
+            };
+
+            return build(usage, referenceType);
+        }
+
+        const image$ = Rx.Observable.fromPromise(imageResource.getData());
+        const usages$ = image$
+            .flatMap((image) =>
+                Rx.Observable.fromPromise(image.usages.getData()))
+            .flatMap((usages) => usages.map(usage => usage.getData()))
+            .flatMap(Rx.Observable.fromPromise)
+            .toArray()
+            .map((usages) => {
+
+                const usagesList =
+                    Immutable.fromJS(usages).map(usage =>
+                        usage.set('title', usageTitle(usage)));
+
+                const groupedByState = usagesList.groupBy(usage =>
+                    usage.get('status'));
+
+                const filterByPlatform = (platform) =>
+                    usagesList.filter(usage =>
+                        usage.get('platform') == platform);
+
+                return {
+                    usages: usagesList.toJS(),
+                    groupedByState: groupedByState.toJS(),
+                    hasPrintUsages: !filterByPlatform('print').isEmpty(),
+                    hasDigitalUsages: !filterByPlatform('digital').isEmpty()
+                };
+
+            });
+
+        return usages$
+
+    }
+
+    return image => forImage(image);
```

Comment:
I'm personally not too keen on this pattern we've used elsewhere (the image service) of having a big helper that computes a set of miscellaneous useful values and returns them all. It's less granular which makes it harder to track which is used where (e.g. to know what to test or adapt when changing the service), and it's less performant (everything is computed every time even if only a subset if used).

The alternatives I can think of that we have used so far (around the list and ordered set collection/streams used for results etc) are to [return a set of streams](https://github.com/guardian/grid/blob/usage-event-stream-lambda/kahuna/public/js/lib/data-structure/ordered-set-factory.js#L46-L48), rather than the stream of sets as above (streams are lazy so only the ones used will be computed), or to define [helper functions](https://github.com/guardian/grid/blob/usage-event-stream-lambda/kahuna/public/js/services/image-list.js) that can be composed to transform the plain original stream. These two approaches can be combined too.

I also find it a little confusing that a "service" ends up being a function, rather than a module that holds more descriptive methods of what it does (`service(foo)` vs `service.getUsages(foo)`). Elsewhere we used "factory" instead to denote "a function that returns a new instance".

Now this is just my opinion so if you disagree or have better suggestions, please share!

And I don't think it's worth spending a lot of time refactoring code that works, so if we want to make some adjustments, it may be fair to do whatever is quickest/easiest so we can ship usage!